frontend/src/components/Game.js

```diff
@@ -1,13 +1,198 @@
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    fetch(`${BACKEND_URL}/game/${gameId}`, options)
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        return response.json().then((e) => {
+          throw new Error(e.detail || "Unknown error");
+        });
+      })
+      .then((game) => {
+        const { player_name, repository_id } = game;
+        setPlayerName(player_name);
+        setRepositoryId(repository_id);
+
+        fetch(`${BACKEND_URL}/repository/${repository_id}/tree`, options)
+          .then((response) => {
+            if (response.ok) {
+              return response.json();
+            }
+            return response.json().then((e) => {
+              throw new Error(e.detail || "Unknown error");
+            });
+          })
+          .then((rootDirectory) => {
+            const { subdirectories } = rootDirectory;
+            setDirectories(subdirectories);
+          })
+          .catch((e) => {
+            const message = "Error occurred: " + e.message;
+            console.log(message);
+            setError(message);
+          });
+      })
+      .catch((e) => {
+        const message = "Error occurred: " + e.message;
+        console.log(message);
+        setError(message);
+      });
+  }, [gameId]);
+
+  const handleClickChoose = (directoryName) => {
+    console.log(`Chose directory: ${directoryName}`);
+    setAnswer(directoryName);
+  };
+
+  const handleSubmit = (answer) => {
+    if (answer) {
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          answer: answer,
+        }),
+      };
+
+      fetch(`${BACKEND_URL}/game/${gameId}`, options)
+        .then((response) => {
+          if (response.ok) {
+            setAnswerSubmitted(true);
+            console.log("Submitted successfully");
+          } else {
+            throw new Error("Failed");
+          }
+        })
+        .catch((error) => {
+          console.log("Error occurred:", error.message);
+        });
+    }
+  };
+
+  const handleReturn = () => {
+    if (pathHistory.length > 0) {
+      const previousPathEntry = pathHistory[pathHistory.length - 1];
+      setDirectories(previousPathEntry.directories);
+      setCurrentPath(previousPathEntry.path);
+      setPathHistory((prevPathHistory) =>
+        prevPathHistory.slice(0, prevPathHistory.length - 1)
+      );
+    }
+  };
 
   return (
     <>
-      <h1>Game</h1>
-      <div>Game id = {state.gameId}</div>
+      <h1 className="title">gitguesser</h1>
+      <div className="form">
+        <div className="info">
+          Player: {playerName}
+          <br />
+          Current path: {currentPath}
+        </div>
+        <h2 className="directories">Directories:</h2>
+        <button className="backButton" onClick={handleReturn}>
+          Back
+        </button>
```

Comment:
Can you make this button disabled when the player is already in the root directory?